src/rules/no-effect-decorator.ts

```diff
@@ -18,20 +35,90 @@ export default ESLintUtils.RuleCreator(docsUrl)<Options, MessageIds>({
       description: 'The createEffect creator function is preferred',
       recommended: 'error',
     },
+    fixable: 'code',
     schema: [],
     messages: {
-      [messageId]: 'The createEffect creator function is preferred.',
+      [noEffectDecorator]: 'The createEffect creator function is preferred.',
+      [noEffectDecoratorSuggest]: 'Remove the `@Effect` decorator.',
     },
   },
   defaultOptions: [],
   create: (context) => {
+    const sourceCode = context.getSourceCode()
+
     return {
-      [effectDecorator](node: TSESTree.TSTypeReference) {
+      [classPropertyWithEffectDecorator](node: EffectDecorator) {
+        const isUsingEffectCreator =
+          isIdentifier(node.parent.value.callee) &&
+          node.parent.value.callee.name === createEffect
+
         context.report({
+          ...(isUsingEffectCreator
+            ? {
+                suggest: [
```

Comment:
One more question that I forgot to ask yesterday.
Could this also be a fix instead of a suggest?

Reply:
I went as a suggest because having `@Effect` and also `createEffect`, the fix (it is, the deletion of `@Effect`) could introduce a break in the expected behavior as, for some reason, they can have different configs setted, as for example:

```
@Effect({ dispatch: true })
readonly test$ = createEffect(() => {
  ...
}, { dispatch: false })
```

That being said, do you think that it would be interesting to add a comment explaining this?

Reply:
I think it can just be a fix. My reasoning is that it's probably just a leftover from a refactor, if both effects are still there it's probably already doing something unwanted 😅